integration_tests/src/main/python/sort_test.py

```diff
@@ -55,6 +55,22 @@ def test_single_orderby(data_gen, order):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).orderBy(order))
 
+@pytest.mark.parametrize('data_gen', single_level_array_gens, ids=idfn)
+@pytest.mark.parametrize('order', [f.col('a').asc(), f.col('a').asc_nulls_first(), f.col('a').asc_nulls_last(),
+                                   f.col('a').desc(), f.col('a').desc_nulls_first(), f.col('a').desc_nulls_last()], ids=idfn)
+def test_single_orderby_on_array(data_gen, order):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).orderBy(order))
+
+@allow_non_gpu('SortExec', 'ShuffleExchangeExec')
```

Comment:
I can look into that. We also have a separate issue tracking order by on multi-level arrays, https://github.com/NVIDIA/spark-rapids/issues/7230. We can enable `ShuffleExchangeExec` in that PR since order by on single level arrays does not depend on `ShuffleExchangeExec` yet. Please correct me if I am wrong. Thanks!

Reply:
This is testing that we fall back to the CPU when we expect it too.  I don't think we want to skip that for this test.